Sources/Sentry/SentryHub.m

```diff
@@ -44,6 +44,7 @@
 @property (nonatomic, strong) SentryCrashWrapper *crashWrapper;
 @property (nonatomic, strong) NSMutableSet<NSString *> *installedIntegrationNames;
 @property (nonatomic) NSUInteger errorsBeforeSession;
+@property (nonatomic) NSMutableArray<id<SentrySessionListener>> * sessionListeners;
```

Comment:
`m`: Will there be multiple listeners in the foreseeable future? If not, maybe we can replace this with a single weak reference cause it simplifies things. If you want to keep the array, maybe it's smarter to replace this with weak references via `[NSHashTable weakObjectsHashTable]` to avoid potential retain cycles. You can check `SentryANRTracker` and `SentryFramesTracker` for reference.